### PR TITLE
[FEATURE] Rewrite bounce handling

### LINKED ISSUE #41: Refactor bounce handling
the returnmail.sh, which get and process the bounces, need to be rewrite, since it's not compatible with 7.6

possible implementation:

- using scheduler job to fetch the mail from mailbox
- Mailbox data can be put in each scheduler job

To be able to fetch an email from a mailbox, [php-imap](http://php.net/manual/en/book.imap.php) extension is a requirement.
Using [Fetch](https://github.com/tedious/Fetch) PHP library could speed up the implementation. But it still needs php-imap extension.

### PR DESCRIPTION
**Pull request by kartolo**
Bounce handling is rewritten, so it's not dependent on fetchmail
anymore. It uses the Fetch library (https://github.com/tedious/Fetch/),
which needs PHP IMAP module.

Todo: check PHP IMAP module and documentation

Fixes #41

### CODE REVIEW

**Comment by kartolo** [comment]
Todo: documentation and remove of old bounce handling script

File: Classes/Scheduler/AnalyzeBounceMail.php [hunk 1]
```diff
@@ -0,0 +1,280 @@
+<?php
+/**
```
Comment (simonschaufi) [inline Classes/Scheduler/AnalyzeBounceMail.php]: could you please remote phpstorm comment block?

File: Classes/Scheduler/AnalyzeBounceMail.php [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+                $this->getDatabaseConnection()->quoteStr($midArray['rtbl'], 'sys_dmail_maillog') . '"' .
+                ' AND mid=' . intval($midArray['mid']) . ' AND response_type=0'
+            );
+
+            // only write to log table, if we found a corresponding recipient record
+            if ($this->getDatabaseConnection()->sql_num_rows($res)) {
+                $row = $this->getDatabaseConnection()->sql_fetch_assoc($res);
+                $midArray['email'] = $row['email'];
+                $insertFields = array(
+                    'tstamp' => time(),
```
Comment (simonschaufi) [inline Classes/Scheduler/AnalyzeBounceMail.php]: use: ```$GLOBALS['EXEC_TIME']```

File: Classes/Scheduler/AnalyzeBounceMail.php [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+     * Return mailServer object or false on error
+     *
+     * @return bool|Server
+     */
+    private function connectMailServer()
+    {
+        // check if we can connect using the given data
+        /** @var Server $mailServer */
+        $mailServer = GeneralUtility::makeInstance(
+            'Fetch\\Server',
```
Comment (simonschaufi) [inline Classes/Scheduler/AnalyzeBounceMail.php]: use ```\Fetch\Server::class```

File: Classes/Scheduler/AnalyzeBounceMailAdditionalFields.php [hunk 4]
```diff
@@ -0,0 +1,156 @@
+<?php
+/**
```
Comment (simonschaufi) [inline Classes/Scheduler/AnalyzeBounceMailAdditionalFields.php]: same here

File: Documentation/Configuration/ConfiguringTheAnalysisOfReturnedMails/Index.rst [hunk 5]
```diff
@@ -18,64 +18,35 @@
 Configuring the analysis of returned mails
 ------------------------------------------
 
-There are probably many ways to configure analysis of returned mails.
-We propose an approach based on open Source program fetchmail. For
-more information about fetchmail, see `The fetchmail home page
-<http://catb.org/~esr/fetchmail/>`_ .
+The analysis of the return mails can now be set in the scheduler job.
+It needs the `PHP-IMAP <http://nl3.php.net/manual/en/book.imap.php>`_
```
Comment (simonschaufi) [inline Documentation/Configuration/ConfiguringTheAnalysisOfReturnedMails/Index.rst]: please don't use nl3 subdomain but just php.net/...

**Comment by kartolo** [comment]
@simonschaufi: thanks for reviewing, your change requests are implemented.

**Comment by simonschaufi** [comment]
you're welcome. I just did a code review, no technical testing.

**Pull request merged**